crates/common/tedge_config/src/tedge_config_cli/new_tedge_config.rs

```diff
@@ -0,0 +1,1412 @@
+//! A collection of tools for reading from and writing to tedge.toml
+use std::borrow::Cow;
+use std::collections::HashMap;
+use std::collections::HashSet;
+use std::fmt;
+use std::hash::Hash;
+use std::num::NonZeroU16;
+use std::sync::Mutex;
+use std::time::Duration;
+
+use camino::Utf8PathBuf;
+use doku::Document;
+use figment::providers::Serialized;
+use figment::util::nest;
+use figment::value::*;
+use figment::*;
+use once_cell::sync::Lazy;
+use once_cell::sync::OnceCell;
+
+use certificate::CertificateError;
+use certificate::PemCertificate;
+
+use crate::ConfigSettingError;
+use crate::ConnectUrl;
+use crate::IpAddress;
+use crate::TEdgeConfigLocation;
+use crate::TemplatesSet;
+
+use super::tedge_config_dto::TEdgeConfigDto;
+
+/// The configuration of thin-edge
+///
+/// This handles fetching values from [TedgeConfigDto](super::tedge_config_dto::TEdgeConfigDto) as well as from the system
```

Comment:
I see there is some hard wrapping for most doc comments, but in some places rows are very short and in some places it looks like they're very wide. So you could some of the comments to some arbitrary threshold, but I don't think either thin-edge or rustdoc mandate it, so it's just a suggesstion. I personally wrap comments to 80 columns and use Rewrap extension for VSCode for automatic wrapping.